core/message.go

```diff
@@ -211,6 +212,21 @@ func (msg Message) Serialize() ([]byte, error) {
 	return proto.Marshal(serializable)
 }
 
+// SerializeOriginal generates a new payload containing all data that can be
+// preserved over shutdown (i.e. no data directly referencing runtime c
+// omponents). The serialized data is based on the original message
+func (msg *Message) SerializeOriginal() ([]byte, error) {
+	serializable := &SerializedMessage{
+		StreamID:     proto.Uint64(uint64(msg.orig.streamID)),
+		PrevStreamID: proto.Uint64(uint64(msg.data.streamID)),
+		Timestamp:    proto.Int64(msg.timestamp.UnixNano()),
+		Data:         msg.orig.payload,
+		Metadata:     msg.orig.Metadata,
+	}
+
+	return proto.Marshal(serializable)
+}
+
```

Comment:
please provide a `func serializeMsg(msg *Messsage) ([]byte, error)` method to remove this duplication